core/domain/user_domain.py

```diff
@@ -526,6 +526,114 @@ def mark_lesson_info_modal_viewed(self) -> None:
         self.has_viewed_lesson_info_modal_once = True
 
 
+class UserGroupDict(TypedDict):
+    """Dictionary representing the UserGroup object."""
+
+    user_group_id: str
+    name: str
+    user_usernames: List[str]
+
+
+class UserGroup:
+    """A domain representation for user group."""
+
+    ALPHANUMERIC_REGEX = r'^[a-zA-Z0-9 ]+$'
+
+    def __init__(
+        self,
+        user_group_id: str,
+        name: str,
+        user_usernames: List[str]
+    ) -> None:
+        """Constructs a UserGroup domain object.
+
+        Args:
+            user_group_id: str. The id of the user group.
+            name: str. The name of the user group.
+            user_usernames: List[str]. The list of user usernames
+                attached to user group.
+        """
+        self.user_group_id = user_group_id
```

Comment:
I'm getting a bit confused now ... in the controiller layer you say user_group_name and not name. Would it be possible to keep things consistent up and down the stack where feasible?

(In general I think "name" is probably fine since we'd typically write `user_group.name`.)

Reply:
Oh, right. I have updated everywhere.